packages/gatsby/src/utils/page-data.ts

```diff
@@ -174,9 +174,14 @@ export async function flush(parentSpan?: Span): Promise<void> {
   )
   writePageDataActivity.start()
 
-  // we process node manifests in this location because we need to add the manifestId to the page data.
-  // We use this manifestId to determine if the page data is up to date when routing. Here we create a map of "pagePath": "manifestId" while processing and writing node manifest files.
-  const nodeManifestPagePathMap = await processNodeManifests()
+  let nodeManifestPagePathMap
+
+  if (pagePaths.size > 0) {
+    // we process node manifests in this location because we need to add the manifestId to the page data.
```

Comment:
Do you think we should add a comment here about why we do this check because of how inc builds cli runs?

Reply:
oh wait you did that, nevermind!